internal/impl/devbox.go

```diff
@@ -280,10 +281,23 @@ func (d *Devbox) RunScript(cmdName string, cmdArgs []string) error {
 		return err
 	}
 
-	cmdWithArgs := append([]string{cmdName}, cmdArgs...)
+	var cmdWithArgs []string
 	if _, ok := d.cfg.Shell.Scripts[cmdName]; ok {
 		// it's a script, so replace the command with the script file's path.
 		cmdWithArgs = append([]string{d.scriptPath(d.scriptFilename(cmdName))}, cmdArgs...)
+	} else {
+		// Arbitrary commands should also run the hooks, so we write them to a file as well. However, if the
+		// command args include env variable evaluations, then they'll be evaluated _before_ the hooks run,
+		// which we don't want. So, one solution is to write the entire command and its arguments into the
+		// file itself, but that may not be great if the variables contain sensitive information. Instead,
+		// we save the entire command (with args) into the DEVBOX_RUN_CMD var, and then the script evals it.
+		err := d.writeScriptFile(arbitraryCmdFilename, d.scriptBody("eval $DEVBOX_RUN_CMD\n"))
+		if err != nil {
+			return err
+		}
+		cmdWithArgs = []string{d.scriptPath(d.scriptFilename(arbitraryCmdFilename))}
+		// TODO: move this env var elsewhere. I will move all the env stuff into a single ComputeEnv() function.
+		pluginEnv = append(pluginEnv, fmt.Sprintf("DEVBOX_RUN_CMD=%s", strings.Join(append([]string{cmdName}, cmdArgs...), " ")))
```

Comment:
is pluginEnv the environment generated _after_ the init-hooks and plugin-hooks have run?

Reply:
no, the stuff in `pluginEnv` comes first. do you have a concern related to ordering?